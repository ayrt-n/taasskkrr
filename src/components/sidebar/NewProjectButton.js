import React from 'react';

function NewProjectButton({ openModal, afterSubmit }) {
  const emptyProject = { title: '' };

  const openNewProjectModal = () => {
    openModal('newProject', {project: emptyProject, callback: afterSubmit})
  };

  return (
    <div className="SidebarItem no-hover">
      <button className="SidebarItem-button" onClick={openNewProjectModal}>
        <svg className="SidebarItem-icon" viewBox="0 0 24 24" style={{width: '1rem', height: '1rem'}}>
          <path fill="currentColor" d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z" />
        </svg>
        Add Project
      </button>
    </div>
  );
}

export default NewProjectButton;
