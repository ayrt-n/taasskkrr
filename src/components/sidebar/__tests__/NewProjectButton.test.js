import React from 'react';
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom'
import NewProjectButton from '../NewProjectButton';

// Set up props
const openModalMock = jest.fn();
const afterSubmitMock = jest.fn();

describe('NewProjectButton component', () => {
  it('renders new project button correctly', () => {
    const { container } = render(
      <NewProjectButton
        openModal={openModalMock}
        afterSubmit={afterSubmitMock}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('calls openModal with correct arguments when clicked', () => {
    render(<NewProjectButton openModal={openModalMock} afterSubmit={afterSubmitMock} />);

    const newProjectButton = screen.getByRole('button', {name: /Add Project/i});
    userEvent.click(newProjectButton);

    expect(openModalMock).toHaveBeenCalledTimes(1);
    expect(openModalMock).toHaveBeenCalledWith(
      'newProject',
      {
        header: 'Add Project',
        project: { title: '' },
        callback: afterSubmitMock
      }
    );
  });
});