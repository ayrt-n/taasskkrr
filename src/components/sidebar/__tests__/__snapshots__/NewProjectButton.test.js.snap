// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjectButton component renders new project button correctly 1`] = `
<div>
  <div
    class="SidebarItem no-hover"
  >
    <button
      class="SidebarItem-button"
    >
      <svg
        class="SidebarItem-icon"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          fill="currentColor"
        />
      </svg>
      Add Project
    </button>
  </div>
</div>
`;
